functions/imagemagick/test/index.test.js

```diff
@@ -31,40 +31,64 @@ requestRetry = requestRetry.defaults({
   retryDelay: 1000,
 });
 
-const BUCKET_NAME = process.env.FUNCTIONS_BUCKET;
-const {BLURRED_BUCKET_NAME} = process.env;
 
 const safeFileName = 'bicycle.jpg';
 const offensiveFileName = 'zombie.jpg';
-
+const BUCKET_NAME = process.env.FUNCTIONS_BUCKET;
+const {BLURRED_BUCKET_NAME} = process.env;
+const blurredBucket = storage.bucket(BLURRED_BUCKET_NAME);
 const cwd = path.join(__dirname, '..');
 
-const blurredBucket = storage.bucket(BLURRED_BUCKET_NAME);
+// Successfully generated images require cleanup.
+let cleanupRequired = false;
 
 describe('functions/imagemagick tests', () => {
-  const startFF = port => {
-    return execPromise(
-      `functions-framework --target=blurOffensiveImages --signature-type=event --port=${port}`,
-      {timeout: 15000, shell: true, cwd}
-    );
-  };
-
-  const stopFF = async ffProc => {
-    try {
-      return await ffProc;
-    } catch (err) {
-      // Timeouts always cause errors on Linux, so catch them
-      if (err.name && err.name === 'ChildProcessError') {
-        const {stdout, stderr} = err;
-        return {stdout, stderr};
+  let startFF, stopFF;
+
+  before(() => {
+    startFF = port => {
+      return execPromise(
+        `functions-framework --target=blurOffensiveImages --signature-type=event --port=${port}`,
+        {timeout: 15000, shell: true, cwd}
+      );
+    };
+
+    stopFF = async ffProc => {
+      try {
+        return await ffProc;
+      } catch (err) {
+        // Timeouts always cause errors on Linux, so catch them
+        if (err.name && err.name === 'ChildProcessError') {
+          const {stdout, stderr} = err;
+          return {stdout, stderr};
+        }
+
+        throw err;
       }
+    };
+  });
+
+  before(async () => {
+    let exists = await storage
+      .bucket(BUCKET_NAME)
+      .file(offensiveFileName)
+      .exists();
+    if (!exists[0]) {
```

Comment:
I'd rather see `if (!exists) {`
by setting
`let exists = ... .exists()[0];`

See example where we declare `exists` to be the first element of the data array:
https://cloud.google.com/nodejs/docs/reference/storage/1.6.x/Bucket#exists

Reply:
Sure, I ended up on the current implementation as a result of my path to realizing exists() returns an array.

Should I use `let [exists] = ... .exists()` instead?

Reply:
Sounds fine.